paddle/fluid/memory/allocation/allocator.h

```diff
@@ -72,17 +77,31 @@ class Allocation {
 
   const platform::Place& place() const { return place_; }
 
-  Allocator* allocator() { return allocator_; }
+  virtual ~Allocation();
+
+ private:
+  std::vector<Allocator*> DecoratedAllocators() const {
+    return static_cast<std::vector<Allocator*>>(decorated_allocators_);
+  }
 
-  void set_allocator(Allocator* allocator) { allocator_ = allocator; }
+  inline void RegisterDecoratedAllocator(Allocator* allocator) {
+    decorated_allocators_.push_back(allocator);
+  }
 
-  virtual ~Allocation();
+  inline void PopDecoratedAllocator() { decorated_allocators_.pop_back(); }
+
+  inline Allocator* TopDecoratedAllocator() {
+    return decorated_allocators_.back();
+  }
 
  private:
-  Allocator* allocator_;
   void* ptr_;
   size_t size_;
   platform::Place place_;
+  framework::InlinedVector<Allocator*, 8> decorated_allocators_;
```

Comment:
Why not use `std::vector` directly? Does there have some different efficient?

Reply:
`std::vector<T>`  would do heap allocation, and increases its capacity in a mode like `0->1->2->4->...`, which may be inefficient because `decorated_allocators_` is a small vector. `InlinedVector` would be more inefficient.